test/integration/targets/azure_rm_postgresqlserver/tasks/main.yml

```diff
@@ -300,6 +300,41 @@
     that:
       - output.changed == false
 
+- name: Create Firewall Rule - second
+  azure_rm_postgresqlfirewallrule:
+    resource_group: "{{ resource_group }}"
+    server_name: postgresqlsrv{{ rpfx }}
+    name: firewallrule{{ rpfx }}second
+    start_ip_address: 172.28.10.136
+    end_ip_address: 172.28.10.138
+
+- name: Gather facts PostgreSQL Firewall Rule
+  azure_rm_postgresqlfirewallrule_facts:
+    resource_group: "{{ resource_group }}"
+    server_name: postgresqlsrv{{ rpfx }}
+    name: firewallrule{{ rpfx }}
+  register: output
+- name: Assert that facts are returned
+  assert:
+    that:
+      - output.changed == False
+      - output.rules[0].id != None
+      - output.rules[0].name != None
+
+- name: Gather facts PostgreSQL Firewall Rule
+  azure_rm_postgresqlfirewallrule_facts:
+    resource_group: "{{ resource_group }}"
+    server_name: postgresqlsrv{{ rpfx }}
+  register: output
+- name: Assert that facts are returned
+  assert:
+    that:
+      - output.changed == False
+      - output.rules[0].id != None
```

Comment:
is this solid enough? always returned in this order?

Reply:
hmm... yes if I understand correctly